Library/ViewModels/WatchProjectViewModel.swift

```diff
@@ -0,0 +1,267 @@
+import KsApi
+import Prelude
+import ReactiveSwift
+import Result
+
+public protocol WatchProjectViewModelInputs {
+  func awakeFromNib()
+  func configure(with project: Project)
+  func projectFromNotification(project: Project?)
+  func saveButtonTapped(selected: Bool)
+  func userSessionEnded()
+  func userSessionStarted()
+  func viewDidLoad()
+}
+
+public protocol WatchProjectViewModelOutputs {
+  /// Emits a boolean to determine whether to create haptics
+  var generateSelectionFeedback: Signal<(), NoError> { get }
+
+  /// Emits a boolean to determine whether to create haptics
+  var generateSuccessFeedback: Signal<(), NoError> { get }
+
+  /// Emits when the login tout should be shown to the user.
+  var goToLoginTout: Signal<(), NoError> { get }
+
+  /// Emits a project.
+  var postNotificationWithProject: Signal<Project, NoError> { get }
+
+  /// Emits the accessibility hint for the star button.
+  var saveButtonAccessibilityValue: Signal<String, NoError> { get }
+
+  /// Emits a boolean that determines if the save button is selected.
+  var saveButtonSelected: Signal<Bool, NoError> { get }
+
+  /// Emits when a contextual Push Notification dialog should be shown.
+  var showNotificationDialog: Signal<Notification, NoError> { get }
+
+  /// Emits when the project has been successfully saved and a prompt should be shown to the user.
+  var showProjectSavedAlert: Signal<Void, NoError> { get }
+}
+
+public protocol WatchProjectViewModelType {
+  var inputs: WatchProjectViewModelInputs { get }
+  var outputs: WatchProjectViewModelOutputs { get }
+}
+
+public final class WatchProjectViewModel: WatchProjectViewModelType,
+WatchProjectViewModelInputs, WatchProjectViewModelOutputs {
+
+  public init() {
+    let viewReady = Signal.merge(self.viewDidLoadProperty.signal, self.awakeFromNibProperty.signal)
+
+    let configuredProject = Signal.combineLatest(
+      self.projectProperty.signal.skipNil(),
+      viewReady
+      )
+      .map(first)
+      .map(cached(project:))
```

Comment:
This is key for the optimistic UI approach to work correctly between discovery and drilled-in view. As soon as the project is configured it checks the cache for its 'starred' value.